src/envoy/http/service_control/handler_impl.cc

```diff
@@ -54,6 +55,26 @@ const Envoy::Http::LowerCaseString kAndroidCertHeader{"x-android-cert"};
 constexpr char JwtPayloadIssuerPath[] = "iss";
 constexpr char JwtPayloadAudiencePath[] = "aud";
 
+std::string CheckErrorNameToRcDetail(absl::string_view rc_detail) {
+  return absl::StrCat("service_control_check_error{", rc_detail, "}");
+}
+
+std::string QuotaErrorNameToRcDetail(absl::string_view rc_detail) {
+  return absl::StrCat("service_control_quota_error{", rc_detail, "}");
+}
+
+struct RcDetailsValues {
+  // The request isn't defined in the service config.
+  const std::string UndefinedRequest =
+      "service_control_bad_request{UNDEFINED_REQUEST}";
```

Comment:
A little confusing. How about `UNDEFINED_OPERATION`, `OPERATION_NOT_MATCHED`, etc.

Reply:
I was trying to not use implementation terms, especially in the view of API Gateway, like what we did in https://github.com/GoogleCloudPlatform/esp-v2/pull/350.

Relatively, `OPERATION` is unknown to customers and they only know requests.

Reply:
Ack